patterns/legacy/squid

```diff
@@ -1,4 +1,4 @@
 # Pattern squid3
 # Documentation of squid3 logs formats can be found at the following link:
 # http://wiki.squid-cache.org/Features/LogFormat
-SQUID3 %{NUMBER:timestamp}\s+%{NUMBER:duration}\s%{IP:client_address}\s%{WORD:cache_result}/%{POSINT:status_code}\s%{NUMBER:bytes}\s%{WORD:request_method}\s%{NOTSPACE:url}\s(%{NOTSPACE:user}|-)\s%{WORD:hierarchy_code}/%{IPORHOST:server}\s%{NOTSPACE:content_type}
```

Comment:
🤔 ah. The way this was originally specified, the intent was clearly to make `user` _not_ capture a literal `-` (which indicates user was not provided), but because `NOTSPACE` matches `-`, this always ended up capturing anyway 🤦‍♀️ . That took me a minute to figure out. Good work in making it capture IFF present in the ecs-v1 pattern.

However, I don't think this change to the legacy patterns is necessary here because it is effectively a no-op, and its presence muddies the history of the legacy patterns.

Reply:
okay than, same for me - took me a few to figure. 
we'll let the next person wonder about this than, hopefully no one will have to.